### DISCUSSION TITLE
Start menu Recommended Section

### DISCUSSION THREAD

**Discussion by oomek** [valinet/ExplorerPatcher#120] [Ideas]
Would your patcher be able to hide the recommended section in Windows 11 start menu so we could have more than 3 rows of icons?

**Comment by valinet** [comment]
I don’t know of any method currently. Although, to be honest, I haven’t studied it as much as I have done with Explorer. Maybe it can be done, idk how the UI is created at all, is it hardcoded etc…

I implemented the stuff I use to mimick the way I used Start in Windows 10. All I cared about was the app list, I did not have pined tiles etc. So far, the patching I do to Start is basic, and there is mo basic way to achieve what you described. For the complicated stuff, as I said, I haven’t looked extensively on it. Although, idk, it will come at some point to Insiders, at which point we can just take the files probably. Microsoft said they are working on it, idk why it takes so long though, how bad can they code it to be so hard to expand it to support this…

The way it looks to me, the current menu is kind of hardcoded for some values.

If I find anything of interest, of course I will add it.

**Reply by Gaurav-Original-ClassicShellTester** [reply]
Btw if you care only about the app list, you can have a minimalist Open Shell menu too and have it open directly to the All Apps list in 7 style. :)) You can start scrolling the app list immediately with up/down arrow keys or hover over it and press a letter to jump to it, or scroll with mouse wheel/touchpad. Compared to the Windows menu, it can use the full height of the screen, a denser, more compact layout and you can control the size of icons in it.

**Comment by oomek** [comment]
To be honest I've lost faith that this section will ever receive a toggle switch after observing that Microsoft is removing up-votes on this feature request on Feedback Hub. I could bet they have wet dreams about spamming us with ads there. Apart from one vague response there is still silence regarding our request. Would be cool if it wasn't heavily hard-coded. I saw a screenshot of a guy that managed to 
remove this section and increase the icons count, but did not share how to do it. Can't find the link now.

**Reply by valinet** [reply]
I remember that as well, he said he just manipulated the UI automation tree in some subsequent version. I don't get the people that do not bother themselves writing some guide when they discover something... then any other person coming after them just basically has to do all the work all over again. Just a waste of time, beyond the initial waste of time generated by the fact that not even this new menu is open source, yet "Microsoft <3 open source" 👎 :/ ...

**Reply by oomek** [reply]
Yes, that, on Winaero, https://winaero.com/windows-11-will-get-an-option-to-disable-the-recommended-section-in-the-start-menu/
Is that screenshot of any help?

**Reply by valinet** [reply]
Honestly, idk... I don't know each and every technology under the sun, unfortunately... I am not familiar with that, but I will look into it... I still don't understand why it was so hard for the OP to also share the way he did it... or for Winaero to ask, rather than just copy-paste something of little value outside of uselessly bragging on reddit...

**Reply by valinet** [reply]
I will ping back in 10 years, Visual Studio has to do 1000TB of updates... I HATE this new "technologies", why do you have to install 1 million TBs of SDKs just to explore some UI tree. 20 years ago: `FindWindowW` etc... easy. /rant

**Reply by oomek** [reply]
Tried to attach the debugger with VS 2019, but not seeing that part of the tree. Installing VS 2022 preview now.

**Reply by valinet** [reply]
I am even farther behind, I don't have the UWP tools installed as I don't use that at all... and even if I install that, what next? I have to do that programatically anyway. And injecting UWP processes is a pain in the ass because you have no console, so the new `printf` is now binary: `exit` when something goes wrong.

**Reply by valinet** [reply]
> 
> 
> Tried to attach the debugger with VS 2019, but not seeing that part of the tree. Installing VS 2022 preview now.

Btw, the guy in the screenshot seemed to be using VS2019... I mean, the icon is from VS2019 as far as I can tell.

**Reply by valinet** [reply]
Anyhow, the tree can be inspected with anumber of tools. The question is how to manipulate it...

**Reply by oomek** [reply]
Sorry, my bad, it was VC2017. 2019 doesn't show up a Live Visual Tree, I must be missing some components.

**Reply by oomek** [reply]
> Anyhow, the tree can be inspected with anumber of tools. The question is how to manipulate it...

No idea mate, I just launched it out of curiosity. I have no experience in debugging foreign processes unfortunately.
